### PR TITLE
Making PID for pivot arm

### PR DESCRIPTION
**Pull request by romallan**
My first attempt at PID. I changed the X Button to invoke the PID command but nothing happened. The k constants are in PID file not Constants temporarily.

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -119,7 +120,7 @@ private void configureButtonBindings() {
     m_coDriverController.button(OIConstants.LEFT_BUMPER).whileTrue(new Intake(IntakeConstants.SCORE_LOW_SPEED));
     m_coDriverController.leftTrigger().whileTrue(new Intake(IntakeConstants.INTAKE_SPEED));
     m_coDriverController.button(OIConstants.A_BUTTON).onTrue(new PivotDown(IntakeConstants.PIVOT_DOWN_SPEED));
```
Comment (minhd-vu) [inline src/main/java/frc/robot/RobotContainer.java]: update this codriver `A` button to use `PivotPidCommand`

File: src/main/java/frc/robot/subsystems/PivotSubsystem.java [hunk 2]
```diff
@@ -22,13 +27,18 @@
...
-public class PivotSubsystem extends SubsystemBase {
+public class PivotSubsystem extends PIDSubsystem {
   /** Creates a new PivotSubsystem. */
 
   private CANSparkMax pivotMotor;
 
   public PivotSubsystem() {
    
+    super(
+       // The controller that the command will use
```
Comment (minhd-vu) [inline src/main/java/frc/robot/subsystems/PivotSubsystem.java]: format this file

File: src/main/java/frc/robot/subsystems/PivotSubsystem.java [hunk 3]
```diff
@@ -40,10 +50,26 @@ public PivotSubsystem() {
...
   }
 
+  @Override
+  protected void useOutput(double output, double setpoint) {
+    // if ((getPivotMotorHeight() > 21) && (setpoint > 21)) {
+    //   pivotMotor.set(0);
+    // } else {
+      SmartDashboard.putNumber("PID Target", setpoint);
+      //tiltLeftMotor.set(-1 * (output + getController().calculate(getMeasurement(), setpoint)));
+      pivotMotor.set(output + getController().calculate(getMeasurement(), setpoint));
```
Comment (minhd-vu) [inline src/main/java/frc/robot/subsystems/PivotSubsystem.java]: remove these comments

**Review by minhd-vu — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/PivotSubsystem.java (hunk 2 above)
Comment (romallan) [reply src/main/java/frc/robot/subsystems/PivotSubsystem.java]: What do you mean by format?

Comment (minhd-vu) [reply src/main/java/frc/robot/subsystems/PivotSubsystem.java]: I think the indentation is off. Do `CTRL+SHIFT+P` and search for `format`

File: src/main/java/frc/robot/Constants.java [hunk 4]
```diff
@@ -50,6 +50,10 @@ public static final class IntakeConstants {
     public static final double PIVOT_MAX = 10;
     public static final double PIVOT_MIN = 0;
 
+    public static final double PIVOT_kP = 0.025;
+    public static final double PIVOT_kI = 0;
+    public static final double PIVOT_kD = 0;
```
Comment (Shiv-Shah) [inline src/main/java/frc/robot/Constants.java:56]: How was all of your values tested. You can use the Hardware Client for help on this

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -109,8 +110,8 @@ private void configureButtonBindings() {
     m_driverController.button(OIConstants.RIGHT_BUMPER).whileTrue(new Intake(IntakeConstants.SCORE_MID_SPEED));
     m_driverController.button(OIConstants.LEFT_BUMPER).whileTrue(new Intake(IntakeConstants.SCORE_LOW_SPEED));
     m_driverController.leftTrigger().whileTrue(new Intake(IntakeConstants.INTAKE_SPEED));
-    m_driverController.button(OIConstants.A_BUTTON).onTrue(new PivotDown(IntakeConstants.PIVOT_DOWN_SPEED));
-    m_driverController.button(OIConstants.X_BUTTON).onTrue(new PivotUp(IntakeConstants.PIVOT_UP_SPEED));
+    m_driverController.button(OIConstants.A_BUTTON).onTrue(new PivotPIDCommand(18));
```
Comment (Shiv-Shah) [inline src/main/java/frc/robot/RobotContainer.java]: Is 18 a setpoint? If not what does it mean. Make it a variable name of what it means

**Review by Shiv-Shah — COMMENTED** [review]
I think Overall great!!. This needs to be tested with PID values using the hardware client to prove that the values aren't an issue. From then test in real code and then you're good.

File: src/main/java/frc/robot/Constants.java (hunk 4 above)
Comment (romallan) [reply src/main/java/frc/robot/Constants.java:56]: I copy pasted those numbers from last year's charged up repo and just tested it in tele-op on Roomba. The arm was very precise and I saw no oscillations in the pivot. But, the encoder values in SmartDashboard started to drift from the arm's actual position as I continued to move the arm up and down in a single test. Kern's said we are using absolute encoders on this year's robot so we shouldn't see the same result.

File: src/main/java/frc/robot/RobotContainer.java (hunk 5 above)
Comment (romallan) [reply src/main/java/frc/robot/RobotContainer.java]: Changed the former constant PIVOT_MAX and MIN to equal our setpoints and used that in the commands

File: src/main/java/frc/robot/Constants.java (hunk 4 above)
Comment (Shiv-Shah) [reply src/main/java/frc/robot/Constants.java:56]: Just a note on the new robot these values may be different and should be different

**Pull request closed without merging**